### ISSUE TITLE
BUG on CC0012: RethrowExceptionAnalyzer throwing "Sequence contains no elements"

### ISSUE THREAD

**Issue by giggio** [code-cracker/code-cracker#226]
The exception is being thrown when CodeCracker runs against corefx (see `test\CSharp\AnalyzeCoreFx.ps1`).
The problem, looking at the log, seems to be on `corefx\src\System.Linq.Parallel`.

`CC0012` is disabled until this is fixed.

**Comment by sqdavid** [comment]
I couldn't rune code-cracker agains corefx (some build issues), but reading the code I build this example that fires the analyzer and throws the exception

```csharp
            var ex = new ArgumentException("An Exception");
            throw ex;
```
It seems that the analyzer assumes every throw expression is inside a catch. I guess just adding an if would solve the issue.
If nobody is working on it I could try it.

**Comment by carloscds** [comment]
@sqdavid Go ahead!

**Comment by giggio** [comment]
Fixed on #232.
